### PR TITLE
Use an environment variable to snapshot the state into

### PR DESCRIPTION
**Pull request by dannywillems**
Using the VM configuration might be better. Let me know what you think.

### CODE REVIEW

File: optimism/src/mips/witness.rs [hunk 1]
```diff
@@ -813,7 +813,12 @@ impl<Fp: Field> Env<Fp> {
 
     fn snapshot_state_at(&mut self, at: &StepFrequency) {
         if self.should_trigger_at(at) {
-            let filename = format!("snapshot-state-{}.json", self.instruction_counter);
+            let snapshot_directory =
+                std::env::var("ZKVM_SNAPSHOT_DIRECTORY").unwrap_or(".".to_string());
```
Comment (rbonichon) [inline optimism/src/mips/witness.rs:817]: There is a command-line argument for that as well (see https://github.com/o1-labs/proof-systems/pull/1523#discussion_r1425375789)

Comment (rbonichon) [reply optimism/src/mips/witness.rs:817]: Well not really for the directory parameter.

**Review by rbonichon — COMMENTED** [review]
Why not but shouldn't this be handled through the `snapshot-fmt` which would recursively create directory on need?

File: optimism/src/mips/witness.rs (hunk 1 above)
Comment (dannywillems) [reply optimism/src/mips/witness.rs:817]: I totally agree, it is far better with the parameter. I don't see how to use it tbh. Why was it added?

Comment (rbonichon) [reply optimism/src/mips/witness.rs:817]: It's a Cannon Cli argument, so we need to be able to parse it 😆

**Comment by dannywillems** [comment]
Closing the PR at the moment. Not a priority.

**Pull request closed without merging**